### ISSUE TITLE
Is there a way to get the quantization and topographical error?

### ISSUE THREAD

**Issue by fciannel** [peterwittek/somoclu#99]
How can we verify that the training process was good enough after it has completed? Is there a way to get the topographical and quantization error per epoch, to see how well the training is progressing?

**Comment by peterwittek** [comment]
Which interface are you using? Can you define topographical and quantization error?

**Comment by fciannel** [comment]
I am using the python interface.

Here are the definitions from the Matlab toolbox:

qe : Average distance between each data vector and its BMU. Measures map
resolution.
te : Topographic error, the proportion of all data vectors for which first
and second BMUs are not adjacent units. Measures topology preservation.

I was running sompy.py and I was having large quantization error in my
setup (around 15), so I wanted to check whether the performance is the same
on somoclu.

BTW great job, the interface is well documented and easy to use!

**Comment by peterwittek** [comment]
Related questions keep popping up, but there is no function implemented to calculate these two quantities. If you want to give it a shot, I would be very happy to merge them.

It is a notch harder to monitor it across epochs. All calculations are pushed to the low-level C++/CUDA code, and the Python interface only gets the results back once all requested epochs are finished. One way to get around it is to request an epoch and manually adjust the learning rate and the radius.

**Comment by fciannel** [comment]
Thanks, I will see if I have time, it depends on how much I use it and need the functionality. For now I am just using it to do visualization for a research I am doing, but I might use it more in the future.

 

But given that we don’t have those indices, how do you measure the performance of the training phase?

**Comment by peterwittek** [comment]
I also use it mainly for visualization, so the scientific way of evaluating the result has always been eyeballing it. Occasionally, I was interested in clustering, and then I measured cluster consistency.

**Comment by fciannel** [comment]
I can eventually add a measure for the errors if not per epoch, at least at
the end of the training, so we have some kind of measure of the performance
of the training.

Thanks a lot for your replies and help.

**Comment by peterwittek** [comment]
That would be great, thanks.

**Comment by akol67** [comment]
> I also use it mainly for visualization, so the scientific way of evaluating the result has always been eyeballing it. Occasionally, I was interested in clustering, and then I measured cluster consistency.

Which metric are you using? WB index? Calinski-Harbusz ?....another one?
